### PR TITLE
Edited docstring, y_pred for SparseCategoricalAcc

### PR DESCRIPTION
**Pull request by jaketae**
1. The `y_pred` example was previously given as `[0.1, 0.9, 0.8]`, which is odd given that the probabilities should sum up to one after passing through softmax. Edited to `[0.1, 0.6, 0.3]` to maintain integrity of example while posing a probable result.
2. The docstring for master was not up-to-date with that of Tag 2.1.0. Made changes so that they are even.

Relates to #36844.

### CODE REVIEW

File: tensorflow/python/keras/metrics.py [hunk 1]
```diff
@@ -769,8 +769,11 @@ def __init__(self, name='categorical_accuracy', dtype=None):
 @keras_export('keras.metrics.SparseCategoricalAccuracy')
 class SparseCategoricalAccuracy(MeanMetricWrapper):
   """Calculates how often predictions matches integer labels.
-
-  You can provide logits of classes as `y_pred`, since argmax of
+  
+  For example, if `y_true` is `[[2], [1]]` and `y_pred` is
```
Comment (pavithrasv) [inline tensorflow/python/keras/metrics.py]: This change can be omitted since the example below conveys the same information.

Comment (jaketae) [reply tensorflow/python/keras/metrics.py]: The issue I referenced asks for better documentation explaining why values are calculated as such. Do you still recommend that I omit the explanation?

Comment (pavithrasv) [reply tensorflow/python/keras/metrics.py]: The sentence here is just putting the below example into words. We are not really explaining how we got the values here ie the formula used etc. This explanation is something I have been meaning to add to metrics soon. We have added such documentation to losses.py already. Please take a look.

Comment (jaketae) [reply tensorflow/python/keras/metrics.py]: Thanks for the feedback. I looked at `losses.py` and understood what you meant. I omitted the textual explanation and wrote an equation in lieu of it, but I'm not sure if this is consistent with the style of the equations in `losses.py`. In particular, I assumed that `==` would be interpreted as an element-wise boolean operator, which is not the case in default Python syntax, nor is it a coherent mathematical expression. Do you have any pointers or suggestions? Thank you.

Comment (pavithrasv) [reply tensorflow/python/keras/metrics.py]: You can use  np.equal in place of that. Also weighting denom should be sum of sample_weights, not n_pred. Thank you!

Comment (jaketae) [reply tensorflow/python/keras/metrics.py]: Thanks for the pointer. I was originally thrown off by the note that `sample_weight` defaults to 1, which is why I thought division by `n_pred` is necessary (I mistakenly thought that the weights would default to something like `[1, 1]`, which isn't true).

However, after some more thought, I'm wondering if we need a weighting denominator at all, given that the sum of `sample_weight` is always going to be 1. In other words, would `acc = np.dot(sample_weight, np.equal(y_true, np.argmax(y_pred, axis=1))` suffice? Thanks again!

Comment (jaketae) [reply tensorflow/python/keras/metrics.py]: @pavithrasv I pushed a commit based on your previous feedback. If time permits, I'd appreciate your review! Thanks in advance.

File: tensorflow/python/keras/metrics.py [hunk 2]
```diff
@@ -770,6 +770,8 @@ def __init__(self, name='categorical_accuracy', dtype=None):
 class SparseCategoricalAccuracy(MeanMetricWrapper):
   """Calculates how often predictions matches integer labels.
 
+  `acc = np.dot(sample_weight, np.equal(y_true, np.argmax(y_pred, axis=1))`
+
```
Comment (mihaimaruseac) [inline tensorflow/python/keras/metrics.py]: I would recommend the following

```suggestion
  ```python
  acc = np.dot(sample_weight, np.equal(y_true, np.argmax(y_pred, axis=1))
  ```
```

**Review by mihaimaruseac — CHANGES_REQUESTED** [review]

File: tensorflow/python/keras/metrics.py (hunk 2 above)
Comment (jaketae) [reply tensorflow/python/keras/metrics.py]: Thank you for the feedback. I have incorporated the changes into the PR.

**Review by mihaimaruseac — APPROVED** [review]
